server/src/main/java/org/apache/druid/curator/BoundedExponentialBackoffRetryWithQuit.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.curator;
+
+import org.apache.curator.RetrySleeper;
+import org.apache.curator.retry.BoundedExponentialBackoffRetry;
+import org.apache.druid.java.util.common.logger.Logger;
+
+import java.util.function.Function;
+
+public class BoundedExponentialBackoffRetryWithQuit extends BoundedExponentialBackoffRetry
+{
+
+  private static final Logger log = new Logger(BoundedExponentialBackoffRetryWithQuit.class);
+
+  private final Function<Void, Void> exitFunction;
+
+  public BoundedExponentialBackoffRetryWithQuit(Function<Void, Void> exitFunction, int baseSleepTimeMs, int maxSleepTimeMs, int maxRetries)
```

Comment:
Line longer than 120 cols.

Reply:
Why using `Function<Void, Void>` rather than Runnable?

Reply:
Fixed line length.

> Why using Function<Void, Void> rather than Runnable?
 
Initially, I used a Callable, but discovered that the method signature included a "throws" which played badly with the allowRetry method signature.  I switched to function because it doesn't suffer from that.  I don't want to use Runnable because it would lead people to believe that it needs to be executed from a Thread, which it does not.

Reply:
To me, `Function<Void, Void>` is more confusing. Could you please define an interface like "ExitAction"?

Reply:
I switched it to your original choice of Runnable rather than to write an interface that looked identical to Runnable/Callable.